merge

sserhii-95 opened the pull request.

Hronom committed:
Added support for @DecimalMin, @DecimalMax annotations

mbknor committed:
Fixing 'Add  processing of polymorphism info for CUSTOM id if the JsonIdResolver is present' so it works for Scala 2.10 and 2.11

mbknor committed:
Fixes #78 - Added support for JsonTypeInfo.Id.MINIMAL_CLASS

mbknor committed:
Added tests for 'Added support for @DecimalMin, @DecimalMax annotations'

mbknor committed:
Setting version to 1.0.31

mbknor committed:
Setting version to 1.0.32-SNAPSHOT

luqasn committed:
Fixes #87 Now generating correct name for inner classes with JsonTypeInfo.Id.MINIMAL_CLASS

Other improvement:
uses TypeIdResolver from Jackson
in order to stay closer to the original id generation

mbknor committed:
Setting version to 1.0.32

mbknor committed:
Setting version to 1.0.33-SNAPSHOT

ebolwidt-atlassian committed:
Provide option to ignore unknown/additional properties for objects

mbknor committed:
Updating scala-versions

mbknor committed:
Ref #90 - added test for additionalProperties / failOnUnknownProperties

mbknor committed:
Setting version to 1.0.33

mbknor committed:
Setting version to 1.0.34-SNAPSHOT

abersnaze committed:
add support for the examples annotation.

luqasn committed:
Add support for generic types with varying type arguments

mbknor committed:
Added info about @JsonSchemaExamples to README

mbknor committed:
Bumping dependency versions

mbknor committed:
Setting version to 1.0.34

mbknor committed:
Setting version to 1.0.35-SNAPSHOT

ebolwidt-atlassian committed:
Square brackets are invalid in the fragment section of a URI - changed them to round parentheses

mbknor committed:
Changing from oraclejdk8 to openjdk8 to fix travis-build

mbknor committed:
Bumped versions

mbknor committed:
It is now possible to use Kotlin in tests

mbknor committed:
Ref #97 Added Kotlin-testcase for the issue. Removed regression, but still ignores kotlin-default-values when resolving required

mbknor committed:
Fixes #99 Using @JsonSchemaInject(merge=false) on top-level class should stop rendering the rest of the class

mbknor committed:
Setting version to 1.0.35

mbknor committed:
Setting version to 1.0.36-SNAPSHOT

mbknor committed:
Ref #101 Added 'Automatic-Module-Name' to META-INF/MANIFEST.MF to fix JDK9+-issue

mbknor committed:
Setting version to 1.0.36

mbknor committed:
Setting version to 1.0.37-SNAPSHOT

mbknor committed:
Preventing that pattern written by @Pattern is not overwritten by @NotBlank

mbknor committed:
Fixes #111 - Added support for validation-groups

mbknor committed:
Replaced links in README from (no longer maintained) jdorn/json-editor to json-editor/json-editor

mbknor committed:
Added support for the @Email-annotation on javax.validation.constraints

mbknor committed:
Fixes #96 Added support for JsonSchema DRAFT-06, DRAFT-07 and DRAFT-2019-09

mbknor committed:
Setting version to 1.0.37

mbknor committed:
Setting version to 1.0.38-SNAPSHOT

mbknor committed:
Small cleanup

mbknor committed:
Fixes #117 - Now default-value read from @JsonProperty(.., defaultValue) is prefered over @JsonSchemaDefault

mbknor committed:
Setting version to 1.0.38

mbknor committed:
Setting version to 1.0.39-SNAPSHOT

Commit:
use jackson @JsonTypeName for schema definition if defined

Commit:
minor code cleanup.

Commit:
using @JsonTypeName annotation name for generic types too if defined.

mbknor committed:
Small changes to PR

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/sserhii-95/mbknor-jackson-jsonSchema/79d5cf61485e8f7a98a754d3600e9cfa303bb4be/sserhii-95/mbknor-jackson-jsonSchema/e370f80d5dd20eb9396455ab2ddfd7083d0e25fb/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) your code in 7.106 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/sserhii-95/mbknor-jackson-jsonSchema/79d5cf61485e8f7a98a754d3600e9cfa303bb4be/sserhii-95/mbknor-jackson-jsonSchema/e370f80d5dd20eb9396455ab2ddfd7083d0e25fb/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=sserhii-95)

mbknor committed:
Setting version to 1.0.39

mbknor committed:
Setting version to 1.0.40-SNAPSHOT

mbknor committed:
JsonSchemaInject now works when injecting stuff into genrrated jackson-nodes - Fixes #122

mbknor committed:
Upgrading to jackson 2.11.0

big-andy-coates committed:
Add support for @JsonTypeInfo on interfaces

fixes: https://github.com/mbknor/mbknor-jackson-jsonSchema/issues/134

Jackson supports `@JsonTypeInfo` on interfaces, not just classes. If `use` is set to anything other than `Id.Type` then currently the generator invokes `ClassGraph` to find the the subtypes of the type. However, `ClassGraph` has different methods for finding implementations of an interface vs subtypes of a class: `getClassesImplementing` vs `getSubclasses` respectively.

This change enhances the generator to call `getClassesImplementing` when the type annotated with `@JsonTypeInfo` is an interface.

mbknor committed:
Upgrading to jackson 2.12.1

mbknor committed:
Upgrading Scala-versions - dropping support for scala 2.10

mbknor committed:
Merge branch 'big-andy-coates-interface_support'
